packages/ai/src/stream.ts

```diff
@@ -120,6 +120,9 @@ function mapOptionsForApi<TApi extends Api>(
 		maxTokens: options?.maxTokens || Math.min(model.maxTokens, 32000),
 		signal: options?.signal,
 		apiKey: apiKey || options?.apiKey,
+		// Pass through OAuth context fields
+		headers: options?.headers,
```

Comment:
Headers and baseUrl are fetched from the passed in Model, no need to pass them in here.